addons/sale/models/account_move.py

```diff
@@ -154,6 +154,43 @@ def _is_downpayment(self):
         self.ensure_one()
         return self.line_ids.sale_line_ids and all(sale_line.is_downpayment for sale_line in self.line_ids.sale_line_ids) or False
 
+    def _get_amount_from_sales_order(self, order):
+        """
+        Consider all lines on any invoice in self that stem from the sales order `order`. (All those invoices belong to order.company_id)
+        This function returns the sum of the totals of all those lines.
+        Note that this amount may be bigger than `order.amount_total`.
+        """
+        order_amount = 0
+        for invoice in self:
+            prices = sum(invoice.line_ids.filtered(lambda x: order in x.sale_line_ids.order_id).mapped('price_total'))
+            order_amount += invoice.currency_id._convert(
+                prices * -invoice.direction_sign,
+                order.currency_id,
+                invoice.company_id,
+                invoice.date,
+            )
+        return order_amount
+
+    def _get_partner_credit_warning_exclude_amount(self):
+        # EXTENDS module 'account'
+        # Consider the warning on a draft invoice created from a sales order.
+        # After confirming the invoice the (partial) amount (on the invoice)
+        # stemming from sales orders will be substracted from the credit_to_invoice.
+        # This will reduce the total credit of the partner.
+        # The computation should reflect the change of credit_to_invoice from 'res.partner'.
+        # (see _compute_credit_to_invoice and _compute_amount_to_invoice from 'sale.order' )
+        exclude_amount = super()._get_partner_credit_warning_exclude_amount()  # self.ensure_one()
+        for order in self.line_ids.sale_line_ids.order_id:
```

Comment:
I did it like this to mirror the way `order.amount_to_invoice` is computed (and currency is converted in _compute_credit_to_invoice).
Since confirming the invoice would update the `order.amount_to_invoice` /  partner.`credit_to_invoice` like this.

When just using the total in company's currency I see the following problems:
* We could add lines to the invoice that are not on the sales order
* We could modify lines to invoice more than is left in `order.amount_to_invoice`. So we have iterate over the individual orders anyway to not take more than `order.amount_to_invoice` per order.

Since it is only called for a single invoice the performance is probably not that bad?
But since it is just a warning maybe just taking the total would be acceptable?